comb_spec_searcher/rule_db/base.py

```diff
@@ -55,13 +55,11 @@ def add(self, start: int, ends: Iterable[int], rule: AbstractRule) -> None:
         ends = tuple(sorted(ends))
         if isinstance(rule, VerificationRule):
             self.set_verified(start)
-        if (
-            isinstance(rule, Rule)
-            and len(ends) == 1
-            and rule.constructor.is_equivalence()
-        ):
+        is_equiv = rule.is_equivalence()
+        if is_equiv:
             self.set_equivalent(start, ends[0])
-        self.rule_to_strategy[(start, ends)] = rule.strategy
+        if len(ends) != 1 or is_equiv or not self.are_equivalent(start, ends[0]):
+            self.rule_to_strategy[(start, ends)] = rule.strategy
```

Comment:
This is the only bit I don't understand. What are we really trying to check here? (And should `is_equiv` be `not is_equiv`?)

Reply:
The goal is to only store a rule if it is (a) not an equivalence rule and (b) doesn't have the same start -> ends as an already existing equivalence rule.

Reply:
Okay, I get it now. It feels like an if statement with a lot of logic. Maybe we can do our future-selves a favor and leave a comment?